cnxarchive/tests/views/test_extras.py

```diff
@@ -89,126 +83,190 @@ def test_extras(self, cursor):
+             u'url': u'http://creativecommons.org/licenses/by/2.0/',
+             u'version': u'2.0'},
+            {u'code': u'by-nd',
+             u'isValidForPublication': False,
+             u'name': u'Creative Commons Attribution-NoDerivs License',
+             u'url': u'http://creativecommons.org/licenses/by-nd/2.0',
+             u'version': u'2.0'},
+            {u'code': u'by-nd-nc',
+             u'isValidForPublication': False,
+             u'name': u'Creative Commons Attribution-NoDerivs-NonCommercial License',
+             u'url': u'http://creativecommons.org/licenses/by-nd-nc/2.0',
+             u'version': u'2.0'},
+            {u'code': u'by-nc',
+             u'isValidForPublication': False,
+             u'name': u'Creative Commons Attribution-NonCommercial License',
+             u'url': u'http://creativecommons.org/licenses/by-nc/2.0',
+             u'version': u'2.0'},
+            {u'code': u'by-sa',
+             u'isValidForPublication': False,
+             u'name': u'Creative Commons Attribution-ShareAlike License',
+             u'url': u'http://creativecommons.org/licenses/by-sa/2.0',
+             u'version': u'2.0'},
+            {u'code': u'by',
+             u'isValidForPublication': False,
+             u'name': u'Creative Commons Attribution License',
+             u'url': u'http://creativecommons.org/licenses/by/3.0/',
+             u'version': u'3.0'},
+            {u'code': u'by',
+             u'isValidForPublication': True,
+             u'name': u'Creative Commons Attribution License',
+             u'url': u'http://creativecommons.org/licenses/by/4.0/',
+             u'version': u'4.0'},
+            {u'code': u'by-nc-sa',
+             u'isValidForPublication': True,
+             u'name': u'Creative Commons Attribution-NonCommercial-ShareAlike License',
+             u'url': u'http://creativecommons.org/licenses/by-nc-sa/4.0/',
+             u'version': u'4.0'}
+            ])
+
+    def assert_subjects_match(self, extras):
+        self.assertEqual(extras['subjects'], [
+            {u'id': 1, u'name': u'Arts',
+             u'count': {u'module': 0, u'collection': 0},
+             },
+            {u'id': 2, u'name': u'Business',
+             u'count': {u'module': 0, u'collection': 0},
+             },
+            {u'id': 3, u'name': u'Humanities',
+             u'count': {u'module': 0, u'collection': 0},
+             },
+            {u'id': 4, u'name': u'Mathematics and Statistics',
+             u'count': {u'module': 7, u'collection': 1},
+             },
+            {u'id': 5, u'name': u'Science and Technology',
+             u'count': {u'module': 6, u'collection': 1},
+             },
+            {u'id': 6, u'name': u'Social Sciences',
+             u'count': {u'module': 0, u'collection': 0},
+             }
+            ])
+
+    def assert_languages_and_count_match(self, extras):
+        self.assertCountEqual(extras['languages_and_count'], [[u'da', 1], [u'en', 19]])
```

Comment:
This should ensure the lists have the same elements and the same counts of each element, in any order. This query does not guarantee the order of the languages.